atkinson/errors/dlrn.py

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env python
+"""
+Atkinson Dlrn Error Handling.
+
+The error classes here are for errors that occur when interacting with dlrn
+and not errors in the atkinson applicaiton itself.
+"""
+import datetime
+import os.path
+
+from atkinson.errors import BaseError
+
+
+class DlrnFTBFSError(BaseError):
+    """ DLRN Failing To Build From Source Error handler """
+    def __init__(self, release, config, reporter, error_id=None):
+        self.__release = release
+        self.__conf = config
+        self.__packages = None
+        self.__report = None
+        self.__reporter = reporter
+
+        if error_id:
+            self.__report = self.__reporter.get(error_id, config)
+        else:
+            title = self.__conf.get('title') + f"[{release}]"
+            self.__report = self.__reporter.new(title,
+                                                self.message,
+                                                self.__conf)
+
+    @property
+    def id(self):
+        """ Return the id for the error """
+        return self.__report.report_id
+
+    @property
+    def packages(self):
+        """ The list packages that failed to build """
+        return self.__packages
+
+    @packages.setter
+    def packages(self, packages):
+        self.__packages = {'Failing Builds': packages}
+
+    @property
+    def message(self):
+        """ The error massage to display """
+        url = os.path.join(self.__conf.get('url', ''), 'status_report.html')
+        message = (f"Results as of: {datetime.datetime.utcnow()} UTC\n"
+                   f"Build Details: {url}")
+        return message
+
+    def action(self):
+        """ Run reporting actions """
+
+        if self.__report is None:
+            self.__report = self.__reporter.new(self.__conf.get('title'),
+                                                self.message,
+                                                self.__conf)
```

Comment:
Shouldn't this already be handled by the if/else in the init? Maybe the init check should just be this on in here?

Reply:
Good catch, this was dead code that I never pruned out after a refactor.